File: src/main/java/org/logister/android/LogisterResponse.java
package org.logister.android;

/** Response returned by Logister after an ingest request. */
public final class LogisterResponse {
    private final int statusCode;
    private final String body;

    public LogisterResponse(int statusCode, String body) {
        this.statusCode = statusCode;
        this.body = body == null ? "" : body;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public String getBody() {
        return body;
    }

    public boolean isAccepted() {
        return statusCode >= 200 && statusCode < 300;
    }
}

File: src/main/kotlin/org/logister/android/LogisterResponse.kt
package org.logister.android

/** Response returned by Logister after an ingest request. */
public class LogisterResponse @JvmOverloads constructor(
    public val statusCode: Int,
    body: String? = "",
    public val isQueued: Boolean = false,
) {
    public val body: String = body ?: ""
    public var retryAfterMillis: Long? = null
        private set

    public val isDropped: Boolean
        get() = !isQueued && statusCode == 0 && body.startsWith(DROPPED_PREFIX)

    public val isAccepted: Boolean
        get() = !isQueued && statusCode in 200..299

    public companion object {
        internal fun queued(): LogisterResponse = LogisterResponse(0, "queued for retry", true)
        internal fun withRetryAfter(statusCode: Int, body: String?, retryAfterMillis: Long?): LogisterResponse =
            LogisterResponse(statusCode, body).also { it.retryAfterMillis = retryAfterMillis }

        internal fun dropped(reason: String): LogisterResponse =
            LogisterResponse(0, "$DROPPED_PREFIX$reason")

        internal fun collectionDisabled(): LogisterResponse = dropped("collection disabled")

        private const val DROPPED_PREFIX: String = "dropped: "
    }
}
